homeassistant/components/nexia/sensor.py

```diff
@@ -0,0 +1,324 @@
+                    percent_conv,
+                )
+            )
+
+        #######################################################################
+        # Zone Sensors
+        #######################################################################
+        for zone_id in thermostat.get_zone_ids():
+            zone = thermostat.get_zone_by_id(zone_id)
+            unit = (
+                TEMP_CELSIUS
+                if thermostat.get_unit() == UNIT_CELSIUS
+                else TEMP_FAHRENHEIT
+            )
+            ###################################################################
+            # Temperature
+            entities.append(
+                NexiaZoneSensor(
+                    coordinator,
+                    zone,
+                    "get_temperature",
+                    "Temperature",
+                    DEVICE_CLASS_TEMPERATURE,
+                    unit,
+                    None,
+                )
+            )
+            ###################################################################
+            # Zone Status
+            entities.append(
+                NexiaZoneSensor(
+                    coordinator, zone, "get_status", "Zone Status", None, None,
+                )
+            )
+            ###################################################################
+            # Setpoint Status
+            entities.append(
+                NexiaZoneSensor(
+                    coordinator,
+                    zone,
+                    "get_setpoint_status",
+                    "Zone Setpoint Status",
+                    None,
+                    None,
+                )
+            )
+
+    async_add_entities(entities, True)
+
+
+def percent_conv(val):
+    """Convert an actual percentage (0.0-1.0) to 0-100 scale."""
+    return val * 100.0
+
+
+class NexiaSensor(Entity):
+    """Provides Nexia sensor support."""
+
+    def __init__(
+        self,
+        coordinator,
+        device,
+        sensor_call,
+        sensor_name,
+        sensor_class,
+        sensor_unit,
+        modifier=None,
+    ):
+        """Initialize the sensor."""
+        self._coordinator = coordinator
+        self._device = device
+        self._call = sensor_call
+        self._sensor_name = sensor_name
+        self._class = sensor_class
+        self._state = None
+        self._unit_of_measurement = sensor_unit
+        self._modifier = modifier
+
+    @property
+    def unique_id(self):
+        """Return the unique id of the sensor."""
+        return f"{self._device.thermostat_id}_{self._call}"
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._device.get_name() + " " + self._sensor_name
+
+    @property
+    def device_state_attributes(self):
+        """Return the device specific state attributes."""
+        return {
+            ATTR_ATTRIBUTION: ATTRIBUTION,
+            ATTR_THERMOSTAT_ID: self._device.thermostat_id,
```

Comment:
What would be the use of adding this to the device state attributes ? Things are already grouped by device in the frontend.

Reply:
I thought I removed all of these.  Fixing.

Reply:
Missed it in the zone as well. 

Killing both